addons/l10n_it_edi_ndd/models/account_move.py

```diff
@@ -0,0 +1,53 @@
+from odoo.addons.l10n_it_edi_ndd.models.account_payment_methode_line import L10N_IT_PAYMENT_METHOD_SELECTION
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_payment_method = fields.Selection(
+        selection=L10N_IT_PAYMENT_METHOD_SELECTION,
+        compute='_compute_l10n_it_payment_method',
+        store=True,
+        readonly=False,
+    )
+
+    l10n_it_document_type = fields.Many2one(
+        comodel_name='l10n_it.document.type',
+        compute='_compute_l10n_it_document_type',
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends('line_ids.matching_number', 'payment_state')
+    def _compute_l10n_it_payment_method(self):
+        move_lines_per_matching_number = self.env['account.move.line'].search([
+            ('matching_number', 'in', self.line_ids.filtered('matching_number').mapped('matching_number'))
+        ]).grouped('matching_number')
+
+        for move in self:
+            matching_numbers = move.line_ids.filtered('matching_number').mapped('matching_number')
+            if matching_numbers and move.payment_state in {'in_payment', 'partial'}:
+                # We use matching_numbers[0] directly, assuming there's a valid key in the dictionary.
+                matching_lines = move_lines_per_matching_number.get(matching_numbers[0])
+                if matching_lines and matching_lines.payment_id:
+                    payment_method_line = matching_lines.payment_id.payment_method_line_id[0]
+                    if payment_method_line:
+                        move.l10n_it_payment_method = payment_method_line.l10n_it_payment_method
+                        continue  # Skip to the next move
+
+            # Default handling if no valid matching lines found or if conditions don't match
+            move.l10n_it_payment_method = move.payment_id.payment_method_line_id.l10n_it_payment_method or move.l10n_it_payment_method or 'MP05'
+
+    @api.depends('line_ids')
+    def _compute_l10n_it_document_type(self):
+        for move in self:
+            move.l10n_it_document_type = self.env.ref(f'l10n_it_edi_ndd.{move._l10n_it_edi_get_document_type()}', raise_if_not_found=False)
+
+    def _l10n_it_edi_get_values(self, pdf_values=None):
+        # EXTENDS 'l10n_it_edi'
+        res = super()._l10n_it_edi_get_values(pdf_values)
+        res['document_type'] = self.l10n_it_document_type.code
```

Comment:
```suggestion
        res['document_type'] = self.l10n_it_document_type.id
```
Couldn't we simply use the `id` instead of adding a column to the table?

Reply:
No because the ID will be a number, and i don't think adding a function to get the xml_id from the id of the record is worth it ?

Reply:
Maybe I'm missing something, but are you sure `id` is a number?

![image](https://github.com/user-attachments/assets/511b9566-dccc-4ee1-ae2b-483251c7a06b)

I just find it weird to have the same data repeated in 3 columns :sweat_smile: